ios/MattermostShare/ShareViewController.swift

```diff
@@ -31,6 +32,44 @@ class ShareViewController: SLComposeServiceViewController {
 
     title = Bundle.main.displayName
     placeholder = "Write a message..."
+
+    let config = getManagedConfig()
+    if let inAppPinCode = config["inAppPinCode"] as? String, inAppPinCode == "true" {
+      self.auth(vendor: config["vendor"] as? String)
+    } else {
+      self.loadData()
+    }
+  }
+
+  func auth(vendor: String?) {
+    let context = LAContext()
+
+    var error: NSError?
+    if !context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
+      if let error = error, error.code == kLAErrorPasscodeNotSet {
+        self.showErrorMessage(
+          title: "",
+          message: "This device must be secured with a passcode to use Mattermost.\n\nGo to Settings > Face ID & Passcode.",
+          VC: self
+        )
+      } else {
+        self.showErrorMessage(title: "", message: "Unable to authenticate device owner for Mattermost", VC: self)
+      }
+
+      return
+    }
+
+    let reason = "Secured by " + (vendor ?? "Mattermost")
+    context.evaluatePolicy(.deviceOwnerAuthentication, localizedReason: reason) { success, error in
+      if success {
+        self.loadData()
+      } else {
+        self.showErrorMessage(title: "", message: "Unable to authenticate device owner for Mattermost", VC: self)
+      }
+    }
+  }
+
+  func loadData() {
     entities = store.getEntities(true) as [AnyHashable:Any]?
     sessionToken = store.getToken()
     serverURL = store.getServerUrl()
```

Comment:
answers:
1. this is controlled by the OS and is set in the Share Extension Info.plist file
![image](https://user-images.githubusercontent.com/6757047/58442202-e53e9d80-80b6-11e9-8404-c81383bada37.png)

2. No we don't have a plan for that as it will be very involved adding those native strings to the translation server.